release 0.0.3

francescofarina opened the pull request:
## Proposed changes

Preparing for the new release.

This requires to first merge #68 #72 #81 and mlx new release (including the fast scatter op). 

Close #88 

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

francescofarina commented:
building docs is currently broken for some reason - I'll look into it once we merge #85 #87 into main as that may solve it.

francescofarina commented:
Building docs is now fixed - @TristanBilot whenever we need to import optional packages (that are not specified in the requirements) let's do it in the functions where that's necessary and not in the import statements at the beginning (see how I modified qm7b.py)

TristanBilot commented on benchmarks/framework_comparison/training_loop/launcher.py:
```diff
@@ -0,0 +1,142 @@
+DATASETS = [
+    # "BZR_MD",
+    # "MUTAG",
+    # "DD",
```
shall we uncomment these? (frameworks + datasets)

TristanBilot commented on benchmarks/framework_comparison/training_loop/launcher.py:
```diff
@@ -0,0 +1,142 @@
+print(f"\nmlx version: {version('mlx')}")
+print(f"\nmlx-graphs version: {version('mlx_graphs')}")
+print(f"\ntorch version: {version('torch')}")
+print(f"\ntorch_geometric version: {version('torch_geometric')}")
```
we should add the DGL version here too.

TristanBilot commented on benchmarks/profiling/profile_training.py:
```diff
@@ -0,0 +1,123 @@
+    loss_sum = 0.0
+    for graph in loader:
+        loss = step(graph)
+        # mx.eval(model.parameters(), optimizer.state)
```
remove this comment unless we use `mx.compile` in this script.

TristanBilot commented on README.md:
```diff
@@ -37,15 +33,73 @@ pip install mlx-graphs
+We provide some notebooks to practice `mlx-graphs`.
+
+- Quickstart guide: [notebook](https://mlx-graphs.github.io/mlx-graphs/tutorials/quickstart.html)
+- Graph classification guide: [notebook](https://mlx-graphs.github.io/mlx-graphs/tutorials/graph_classification.html)
```
I made the assumption that those URLs will work, can you confirm they are right @francescofarina?

TristanBilot commented:
Ah also, can you please run the `graph_classification.ipynb` notebook on your M3 Pro? The prints are not finished in the notebook and the results will be better than on my M1 Pro

TristanBilot commented:
Sorry I meant the `benhcmark_pyg_dgl_mxg.ipynb` notebook!

francescofarina commented:
Done! That notebook is not in the docs though (and the actual benchmark is in the benchmarks folder).

TristanBilot commented:
I think it would be worth it to add it to the docs no?

francescofarina commented:
> I think it would be worth it to add it to the docs no?

We can add it but it could be worth adding some brief commentary to it (and removing the compile step), wdyt? The way I see it useful in the docs is if we use it as a way to show the difference wrt pyg or dgl in terms of syntax/usage.

TristanBilot commented:
Actually for the `raph_classification.ipynb` it may be better to keep the last version as the accuracy was higher 😅

TristanBilot commented:
> > I think it would be worth it to add it to the docs no?
> 
> We can add it but it could be worth adding some brief commentary to it (and removing the compile step), wdyt? The way I see it useful in the docs is if we use it as a way to show the difference wrt pyg or dgl in terms of syntax/usage.

I thought this notebook was the "proof" that `mlx-graphs` can be faster than other frameworks, no? It is especially important IMO. Maybe in this release it would be enough to add it as is?

francescofarina commented:
I think notebooks in the docs should be _instructional_. 
Benchmarks should stay in the benchmarks folder in general IMO.

We can maybe add just this one to the docs for now. Let me add some commentary to it.

> Actually for the graph_classification.ipynb it may be better to keep the last version as the accuracy was higher 😅
Same comment as above, mostly instructional - I reduced the epochs from 100 to improve the readability.

francescofarina commented:
@TristanBilot I updated the nb and added it to the docs. I also added a note in the README and the first page of the docs saying that in initial benchmarks we're up to 10x faster than other frameworks - let me know what you think.
